File: part1/task1/task.m
clc;
close all;
clear;

% Main
file_loc = 'data/example_image.mat';
data = load(file_loc);
simple_image_write('image.sim',data);
image_out = simple_image_read('image.sim');

% Plotting at 3 different z-coordinates - [1 5 10]
figure(1)
% subplot(3,1,1)
contourf(image_out.vol(:,:,1))
title('Plot with z=1');

figure(2)
% subplot(3,1,2)
contourf(image_out.vol(:,:,5))
title('Plot with z=5');

figure(3)
% subplot(3,1,3)
contourf(image_out.vol(:,:,10))
title('Plot with z=10');

File: part1/task1/task.py
import scipy.io
from scipy import misc
import imageio

img = scipy.io.loadmat('data/example_image.mat')

def img_load(file_path):

    img = scipy.io.loadmat(file_path)

    print(img)

    # return img,vol,voxdims

# def simple_image_write(img):

#     scipy.io.savemat('data/image.sim', img)


# def simple_image_read():

#     img = imageio.imread('data/image.sim')

#     print(img)
    
# simple_image_write(img)
# simple_image_write(img_load('data/example_image.mat')[0])    
# simple_image_read()
print(img)